src/v/archival/types.cc

```diff
@@ -117,7 +117,21 @@ get_archival_service_config(ss::scheduling_group sg, ss::io_priority_class p) {
 
 bool adjacent_segment_run::maybe_add_segment(
   const cloud_storage::segment_meta& s, size_t max_size) {
-    vlog(archival_log.debug, "Looking at segment meta: {}", s);
+    vlog(
+      archival_log.debug,
+      "{} Segments collected, looking at segment meta: {}, current run meta: "
+      "{}",
+      num_segments,
+      s,
+      meta);
+    if (num_segments == 1 && meta.size_bytes + s.size_bytes > max_size) {
+        // Corner case, we hit a segment which is smaller than the max_size
+        // but it's larger than max_size when combined with its neighbor. In
+        // this case we need to skip the segment.
```

Comment:
nit: "...skip the _previous_ segment" just so it's clear that the intent is to continue considering the current segment

Reply:
done